provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/controller/location/LocationController.java

```diff
@@ -1,3 +1,455 @@
+import gov.va.api.health.providerdirectory.service.controller.Bundler.BundleContext;
+import gov.va.api.health.providerdirectory.service.controller.PageLinks.LinkConfig;
+import gov.va.api.health.providerdirectory.service.controller.Parameters;
+import gov.va.api.health.providerdirectory.service.controller.Validator;
+import gov.va.api.health.stu3.api.resources.Location;
+import gov.va.api.health.stu3.api.resources.OperationOutcome;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import javax.validation.constraints.Min;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Request Mappings for the Location Profile, see
+ * http://www.fhir.org/guides/argonaut/pd/StructureDefinition-argo-location.html for implementation
+ * details.
+ */
+@SuppressWarnings("WeakerAccess")
+@RestController
+@RequestMapping(
+  value = {"/api/Location"},
+  produces = {"application/json", "application/json+fhir", "application/fhir+json"}
+)
+public class LocationController {
+
+  private Transformer transformer;
+
+  private PpmsClient ppmsClient;
+
+  private Bundler bundler;
+
+  private int totalRecords;
+
+  /** Controller setup. */
+  public LocationController(
+      @Autowired LocationController.Transformer transformer,
+      @Autowired Bundler bundler,
+      @Autowired PpmsClient ppmsClient) {
+    this.transformer = transformer;
+    this.bundler = bundler;
+    this.ppmsClient = ppmsClient;
+  }
+
+  private Location.Bundle bundle(MultiValueMap<String, String> parameters, int page, int count) {
+    List<LocationWrapper> root = search(parameters, page, count);
+    LinkConfig linkConfig =
+        LinkConfig.builder()
+            .path("Location")
+            .queryParams(parameters)
+            .page(page)
+            .recordsPerPage(count)
+            .totalRecords(totalRecords)
+            .build();
+    return bundler.bundle(
+        BundleContext.of(linkConfig, root, transformer, Location.Entry::new, Location.Bundle::new));
+  }
+
+  /** Read by identifier. */
+  @GetMapping(value = {"/{publicId}"})
+  public Location readByIdentifier(@PathVariable("publicId") String publicId) {
+    return transformer.apply(search(Parameters.forIdentity((publicId)), 1, 1).get(0));
+  }
+
+  private List<LocationWrapper> search(
+      MultiValueMap<String, String> parameters, int page, int count) {
```

Comment:
One change we started making on `data-query` is not to pass the `page` and `count` around when they are already present in the parameters bundle. You can pass just `parameters` and use this in the methods that need page and count:

```
    int page = Integer.parseInt(parameters.getOrDefault("page", singletonList("1")).get(0));
    int count = Integer.parseInt(parameters.getOrDefault("_count", singletonList("15")).get(0));
```